source/core/Synchronization.cpp

```diff
@@ -751,22 +759,65 @@ VIREO_FUNCTION_SIGNATURE6(QueueRef_Obtain, RefNumValType*, Int32, StringRef, Boo
     }
     return _NextInstruction();
 }
+static void GetQueueRefName(RefNumValType *refnumPtr, StringRef *stringRef) {
+    if (stringRef) {
+        if (refnumPtr) {
+            RefNum realRefnum = QueueRefNumManager::QueueRefManager().LookupAlias(refnumPtr->GetRefNum());
+            QueueRefNumManager::NamedRefNumMapType::iterator it = QueueRefNumManager::QueueRefManager().NamedRefNumMap().begin(), ite = QueueRefNumManager::QueueRefManager().NamedRefNumMap().end();
+            while (it != ite) {
+                if (it->second == realRefnum)
+                    break;
+                ++it;
+            }
+            if (it != ite)
+                (*stringRef)->Type()->CopyData(&it->first,stringRef);
+            else
+                (*stringRef)->Resize1D(0);
+        }
+        else
+            (*stringRef)->Resize1D(0);
+    }
```

Comment:
Shouldn't the else case for if (stringRef) return an empty string, like the other cases.